File: TransCoder_tokenized_test_set_functions/java/NEXT_GREATER_FREQUENCY_ELEMENT.java
static void NFG ( int a [ ], int n, int freq [ ] ) {
  Stack < Integer > s = new Stack < Integer > ( );
  s . push ( 0 );
  int res [ ] = new int [ n ];
  for ( int i = 0;
  i < n;
  i ++ ) res [ i ] = 0;
  for ( int i = 1;
  i < n;
  i ++ ) {
    if ( freq [ a [ s . peek ( ) ] ] > freq [ a [ i ] ] ) s . push ( i );
    else {
      while ( freq [ a [ s . peek ( ) ] ] < freq [ a [ i ] ] && s . size ( ) > 0 ) {
        res [ s . peek ( ) ] = a [ i ];
        s . pop ( );
      }
      s . push ( i );
    }
  }
  while ( s . size ( ) > 0 ) {
    res [ s . peek ( ) ] = - 1;
    s . pop ( );
  }
  for ( int i = 0;
  i < n;
  i ++ ) {
    System . out . print ( res [ i ] + " " );
  }
}

File: TransCoder_tokenized_test_set_functions/cpp/NEXT_GREATER_FREQUENCY_ELEMENT.cpp
void NFG ( int a [ ], int n, int freq [ ] ) {
  stack < int > s;
  s . push ( 0 );
  int res [ n ] = {
    0 };
    for ( int i = 1;
    i < n;
    i ++ ) {
      if ( freq [ a [ s . top ( ) ] ] > freq [ a [ i ] ] ) s . push ( i );
      else {
        while ( freq [ a [ s . top ( ) ] ] < freq [ a [ i ] ] && ! s . empty ( ) ) {
          res [ s . top ( ) ] = a [ i ];
          s . pop ( );
        }
        s . push ( i );
      }
    }
    while ( ! s . empty ( ) ) {
      res [ s . top ( ) ] = - 1;
      s . pop ( );
    }
    for ( int i = 0;
    i < n;
    i ++ ) {
      cout << res [ i ] << " ";
    }
  }